### ISSUE TITLE
[Bug]: Error: Couldn't find any suitable video driver

### ISSUE THREAD

**Issue by kha84** [OpenTTD/OpenTTD#9969]
### Version of OpenTTD

1.10.3, Debian-based ARM64

### Expected result

The game should start

### Actual result

Game doesn't start:

```
pi@orangepi4-lts:~$ openttd
Error: Couldn't find any suitable video driver
```

Tried to add some more debug - it doesn't bring anything new.

```
pi@orangepi4-lts:~$ openttd -d driver=9,grf=9,map=9,misc=9,net=9,sprite=9,oldloader=9,npf=9,yapf=9,freetype=9,script=9,sl=9,gamelog=9,desync=9,console=9    dbg: [misc] /home/pi/.local/share/openttd/ added as search path
dbg: [misc] /home/pi/.openttd/ added as search path
dbg: [misc] /usr/share/games/openttd/ added as search path
dbg: [misc] /home/pi/.config/openttd/ found as config directory
dbg: [misc] /home/pi/.local/share/openttd/ found as personal directory
dbg: [misc] Scanning for tars
dbg: [misc] Scan complete, found 0 files
dbg: [grf] Checking /usr/share/games/openttd/baseset/opengfx/opengfx.obg for base graphics set
dbg: [grf] Adding OpenGFX (7393) as base graphics set
dbg: [grf] Checking /usr/share/games/openttd/baseset/orig_win.obg for base graphics set
dbg: [grf] The file TRG1R.GRF specified in /usr/share/games/openttd/baseset/orig_win.obg is missing
dbg: [grf] The file TRGIR.GRF specified in /usr/share/games/openttd/baseset/orig_win.obg is missing
dbg: [grf] The file TRGCR.GRF specified in /usr/share/games/openttd/baseset/orig_win.obg is missing
dbg: [grf] The file TRGHR.GRF specified in /usr/share/games/openttd/baseset/orig_win.obg is missing
dbg: [grf] The file TRGTR.GRF specified in /usr/share/games/openttd/baseset/orig_win.obg is missing
dbg: [grf] Adding original_windows (0) as base graphics set
dbg: [grf] Checking /usr/share/games/openttd/baseset/orig_dos_de.obg for base graphics set
dbg: [grf] The file TRG1.GRF specified in /usr/share/games/openttd/baseset/orig_dos_de.obg is missing
dbg: [grf] The file TRGI.GRF specified in /usr/share/games/openttd/baseset/orig_dos_de.obg is missing
dbg: [grf] The file TRGC.GRF specified in /usr/share/games/openttd/baseset/orig_dos_de.obg is missing
dbg: [grf] The file TRGH.GRF specified in /usr/share/games/openttd/baseset/orig_dos_de.obg is missing
dbg: [grf] The file TRGT.GRF specified in /usr/share/games/openttd/baseset/orig_dos_de.obg is missing
dbg: [grf] Adding original_dos_de (0) as base graphics set
dbg: [grf] Checking /usr/share/games/openttd/baseset/orig_dos.obg for base graphics set
dbg: [grf] The file TRG1.GRF specified in /usr/share/games/openttd/baseset/orig_dos.obg is missing
dbg: [grf] The file TRGI.GRF specified in /usr/share/games/openttd/baseset/orig_dos.obg is missing
dbg: [grf] The file TRGC.GRF specified in /usr/share/games/openttd/baseset/orig_dos.obg is missing
dbg: [grf] The file TRGH.GRF specified in /usr/share/games/openttd/baseset/orig_dos.obg is missing
dbg: [grf] The file TRGT.GRF specified in /usr/share/games/openttd/baseset/orig_dos.obg is missing
dbg: [grf] Removing original_dos_de (0) as base graphics set (duplicate, lower version)
dbg: [grf] Adding original_dos (1) as base graphics set
dbg: [misc] Loading blitter...
dbg: [driver] Successfully loaded blitter '32bpp-anim'
dbg: [driver] Probing video driver 'sdl' failed with error: No available video device
Error: Couldn't find any suitable video driver
```

In the very same terminal session I'm able to run basic X apps, like xeyes, xclock or any KDE apps - so it has nothing to do with X auth or missing DISPLAY (just seen some other bugs because of that, but the error message was the same)

Looks like the game is able to recognize, that I do have SDL installed:
```
pi@orangepi4-lts:~$ openttd --help
OpenTTD 1.10.3

...
List of video drivers:
               sdl: SDL Video Driver
         dedicated: Dedicated Video Driver
              null: Null Video Driver
...
```

but forceing it to use it doesn't change much:

```
pi@orangepi4-lts:~$ openttd -v sdl -d driver=9,grf=9,map=9,misc=9,net=9,sprite=9,oldloader=9,npf=9,yapf=9,freetype=9,script=9,sl=9,gamelog=9,desync=9,console=9
dbg: [misc] /home/pi/.local/share/openttd/ added as search path
dbg: [misc] /home/pi/.openttd/ added as search path
dbg: [misc] /usr/share/games/openttd/ added as search path
dbg: [misc] /home/pi/.config/openttd/ found as config directory
dbg: [misc] /home/pi/.local/share/openttd/ found as personal directory
dbg: [misc] Scanning for tars
dbg: [misc] Scan complete, found 0 files
dbg: [grf] Checking /usr/share/games/openttd/baseset/opengfx/opengfx.obg for base graphics set
dbg: [grf] Adding OpenGFX (7393) as base graphics set
dbg: [grf] Checking /usr/share/games/openttd/baseset/orig_win.obg for base graphics set
dbg: [grf] The file TRG1R.GRF specified in /usr/share/games/openttd/baseset/orig_win.obg is missing
dbg: [grf] The file TRGIR.GRF specified in /usr/share/games/openttd/baseset/orig_win.obg is missing
dbg: [grf] The file TRGCR.GRF specified in /usr/share/games/openttd/baseset/orig_win.obg is missing
dbg: [grf] The file TRGHR.GRF specified in /usr/share/games/openttd/baseset/orig_win.obg is missing
dbg: [grf] The file TRGTR.GRF specified in /usr/share/games/openttd/baseset/orig_win.obg is missing
dbg: [grf] Adding original_windows (0) as base graphics set
dbg: [grf] Checking /usr/share/games/openttd/baseset/orig_dos_de.obg for base graphics set
dbg: [grf] The file TRG1.GRF specified in /usr/share/games/openttd/baseset/orig_dos_de.obg is missing
dbg: [grf] The file TRGI.GRF specified in /usr/share/games/openttd/baseset/orig_dos_de.obg is missing
dbg: [grf] The file TRGC.GRF specified in /usr/share/games/openttd/baseset/orig_dos_de.obg is missing
dbg: [grf] The file TRGH.GRF specified in /usr/share/games/openttd/baseset/orig_dos_de.obg is missing
dbg: [grf] The file TRGT.GRF specified in /usr/share/games/openttd/baseset/orig_dos_de.obg is missing
dbg: [grf] Adding original_dos_de (0) as base graphics set
dbg: [grf] Checking /usr/share/games/openttd/baseset/orig_dos.obg for base graphics set
dbg: [grf] The file TRG1.GRF specified in /usr/share/games/openttd/baseset/orig_dos.obg is missing
dbg: [grf] The file TRGI.GRF specified in /usr/share/games/openttd/baseset/orig_dos.obg is missing
dbg: [grf] The file TRGC.GRF specified in /usr/share/games/openttd/baseset/orig_dos.obg is missing
dbg: [grf] The file TRGH.GRF specified in /usr/share/games/openttd/baseset/orig_dos.obg is missing
dbg: [grf] The file TRGT.GRF specified in /usr/share/games/openttd/baseset/orig_dos.obg is missing
dbg: [grf] Removing original_dos_de (0) as base graphics set (duplicate, lower version)
dbg: [grf] Adding original_dos (1) as base graphics set
dbg: [misc] Loading blitter...
dbg: [driver] Successfully loaded blitter '32bpp-anim'
Error: Unable to load driver 'sdl'. The error was: No available video device
```

I do have both SDL1 and SDL2 installed and other graphical applications are working just fine:
```
pi@orangepi4-lts:~$ sdl2-config --version
2.0.20
pi@orangepi4-lts:~$ sdl-config --version
1.2.15
```

Just in case the package was installed from one of the official bullseye mirrors:
```
pi@orangepi4-lts:~$ apt policy openttd
openttd:
  Installed: 1.10.3-1
  Candidate: 1.10.3-1
  Version table:
 *** 1.10.3-1 500
        500 http://deb.debian.org/debian bullseye/main arm64 Packages
        500 http://ftp.ru.debian.org/debian bullseye/main arm64 Packages
        500 https://mirror.docker.ru/debian bullseye/main arm64 Packages
        500 http://ftp.by.debian.org/debian bullseye/main arm64 Packages
        500 http://mirror.cloudhosting.lv/debian bullseye/main arm64 Packages
        100 /var/lib/dpkg/status
```

And just on a top of that, trying to use "dedicated" driver just makes it to hang somewhere. Turning all the possible debug on prints something to console periodically but nothing else happens:
```
pi@orangepi4-lts:~$ openttd -v dedicated -d driver=9,grf=9,map=9,misc=9,net=9,sprite=9,oldloader=9,npf=9,yapf=9,freetype=9,script=9,sl=9,gamelog=9,desync=9,console=9
...
dbg: [sl] Saving chunk OBJS
dbg: [sl] Saving chunk PSAC
dbg: [sprite] Compacting sprite cache, inuse=7845352
....
dbg: [sl] Saving chunk OBID
dbg: [sl] Saving chunk OBJS
dbg: [sl] Saving chunk PSAC
dbg: [sprite] Compacting sprite cache, inuse=7845352
```

My system is:
```
pi@orangepi4-lts:~$ neofetch 
       _,met$$$$$gg.          pi@orangepi4-lts 
    ,g$$$$$$$$$$$$$$$P.       ---------------- 
  ,g$$P"     """Y$$.".        OS: Debian GNU/Linux 11 (bullseye) aarch64 
 ,$$P'              `$$$.     Host: OrangePi 4 LTS 
',$$P       ,ggs.     `$$b:   Kernel: 5.18.5-rk3399 
`d$$'     ,$P"'   .    $$$    Uptime: 7 hours, 22 mins 
 $$P      d$'     ,    $$P    Packages: 2044 (dpkg) 
 $$:      $$.   -    ,d$$'    Shell: bash 5.1.4 
 $$;      Y$b._   _,d$P'      Resolution: 1920x1080 
 Y$$.    `.`"Y$$$$P"'         DE: Plasma 5.20.5 
 `$$b      "-.__              WM: KWin 
  `Y$$                        Theme: Adwaita [GTK3] 
   `Y$$.                      Icons: Adwaita [GTK3] 
     `$$b.                    Terminal: konsole 
       `Y$$b.                 Terminal Font: Hack 10 
          `"Y$b._             CPU: (6) @ 1.416GHz 
              `"""            Memory: 1965MiB / 3863MiB 
```

### Steps to reproduce

1. apt install openttd
2. openttd

**Comment by James103** [comment]
1. Does your system have a GPU? If so, what is the make and model of the GPU and what graphics driver version do you have installed?

**Comment by glx22** [comment]
Do other application using sdl works ?
In the `dbg: [driver] Probing video driver 'sdl' failed with error: No available video device` line, `No available video device` directly comes from sdl, so the issue might be sdl itself.

**Comment by kha84** [comment]
> 

Integrated into CPU - it's Mali T860. I have a full fledged KDE running on that machine and everything I was doing I was doing under Konsole on X session in KDE

**Comment by kha84** [comment]
> Do other application using SDL works ? In the `dbg: [driver] Probing video driver 'sdl' failed with error: No available video device` line, `No available video device` directly comes from SDL, so the issue might be SDL itself.

Appreciate if you can give me a hint what other SDL applications are there? Like I can run VLC, I seem it's a SDL app

**Comment by kha84** [comment]
Reg videodrivers. I do have mesa insalled. I checked few examples from the mesa-utils package, like gxgears and gxdemo - they work just fine

**Comment by nielsmh** [comment]
Regardless of anything else, the SDL or SDL2 driver in OpenTTD should be able to work with a plain X server with no acceleration features at all.

Per https://wiki.libsdl.org/FAQUsingSDL you can try setting `SDL_VIDEODRIVER=x11` in the environment to force the driver selection.

It sounds like you have tried this already, but I'll link it too just in case: https://wiki.libsdl.org/FAQLinux#i_get_the_error_no_video_devices_available

**Comment by kha84** [comment]
Appreciate your help and these URLs, @nielsmh 
I tried it out with explicitly setting that - the error message is different this time:
```
pi@orangepi4-lts:~$ SDL_VIDEODRIVER=x11 openttd -v sdl
Error: Unable to load driver 'sdl'. The error was: x11 not available
```
At least this is something new I can probably try to dig out myself. Still confused, how all other X apps including mesa demos are working just fine. I'm starting to think my SDL package was forged somehow. Will definitely need to test some other SDL apps, if I only knew which ones. Went off to dig further

**Comment by glx22** [comment]
You can try `aptitude -v --show-summary=all-packages why libsdl2-2.0-0` to find installed packages using SDL (may need to install `aptitude` first of course). From a quick check I can see `gimp` and `bomberclone` in the list.
With ` apt-rdepends -r libsdl2-2.0-0 | less` you can get packages (installed or not) using SDL.

Oh and your "dedicated" test probably didn't hang, it just started a dedicated server and the output is almost quiet.

**Comment by kha84** [comment]
Awesome, thanks for the hint @glx22 
As it turns out, I have a whole bunch of packages installed, which are dependant on that libsdl2-2.0:
```
  chromium D: xdg-desktop-portal-backend P<- xdg-desktop-portal-backend D: libkf5notifications5 D: phonon4qt5 D: phonon4qt5-backend P<- phonon4qt5-backend D: gstreamer1.0-audiosink P<- gstreamer1.0-audiosink D: libfluidsynth2 D: libsdl2-2.0-0
...
  micropolis D: libsdl-mixer1.2 D: libmikmod3 D: libsdl2-2.0-0
  openttd D: libsdl2-2.0-0
  plasma-discover D: libkf5notifications5 D: phonon4qt5 D: phonon4qt5-backend P<- phonon4qt5-backend D: gstreamer1.0-audiosink P<- gstreamer1.0-audiosink D: libfluidsynth2 D: libsdl2-2.0-0
  plasma-nm D: libkf5notifications5 D: phonon4qt5 D: phonon4qt5-backend P<- phonon4qt5-backend D: gstreamer1.0-audiosink P<- gstreamer1.0-audiosink D: libfluidsynth2 D: libsdl2-2.0-0
...
  supertux D: libsdl2-2.0-0
  systemsettings D: kio D: libkf5notifications5 D: phonon4qt5 D: phonon4qt5-backend P<- phonon4qt5-backend D: gstreamer1.0-audiosink P<- gstreamer1.0-audiosink D: libfluidsynth2 D: libsdl2-2.0-0
```

micropolis (some game) works just fine, KDE applications like mentioned systemsettings, plasma-discover also work fine. Needless to say I'm typing all that in Chromium. BUT, supertux2 is having the very same issue, like the openttd:

```
pi@orangepi4-lts:~$ SDL_VIDEODRIVER=x11 supertux2
[FATAL] /build/supertux-nLN4lO/supertux-0.6.2/src/supertux/main.cpp:603 Unexpected exception: Couldn't initialize SDL: x11 not available
```

So it's indeed something is darn wrong with my SDL, and it's unrelated to openttd for sure. I just need to understand what exactly.

**Comment by nielsmh** [comment]
Would it be an idea to have OpenTTD output some more detailed diagnostics when it fails to start SDL video? Of course that's assuming there even is any more detailed diagnostics that can be gathered.

**Comment by glx22** [comment]
We already use [SDL_GetError()](https://wiki.libsdl.org/SDL_GetError), and I don't think there's a more detailed option.

**Comment by kha84** [comment]
Sorry guys, given I'm almost a zero in debian based distros, can you pls give me a small hand of help or rather advice.

As I said, I'm having a strong feeling that the sdl library I have was somehow compiled without the X11 support, otherwise I have no explanation why it's complaining like that. If I put some random string to SDL_VIDEODRIVER it's complaining on that string the very same way. So to me it's quite an indicator it knows nothing about x11.

This in turn makes me think the deb package which provided that libsdl.so was sideloaded / customly built for the purposes other than I'm trying to use it now (the distro I'm using is some descendant of RetroPie, so prob there's a use of SDL there without X11 support)

I was thinking about reinstalling the package from the official debian repo (given the version of that package is exactly the same), while hoping the guys who built it for debian didn't exclude x11 support for it. For a last resort I'm keeping in mind to rebuild it from deb-src with x11 support myself.

But before I do all that, is there any quick and easy way to crosscheck if libsdl.so file(s) I have in my filesystem are the same (or different) to what will be provided by the package from debian repo? I know of the manual way, like to download that deb package first, unpack it and then diff. This will work for this single package. But I'm also having a wild guess, I might be having much more of such "customized" packages, so to me it makes sense to test everything in one go, to see how bad things are. Off course if there's any option for that among apt/dpkg/aptitude.

**Comment by glx22** [comment]
You can use `aptitude show libsdl2-2.0-0` to get the dependencies, it should probably list `libx11` if it's build with x11 support
And this is the output I get for Ubuntu under WSL2:
```
Package: libsdl2-2.0-0
Version: 2.0.10+dfsg1-3
State: installed
Automatically installed: yes
Multi-Arch: same
Priority: optional
Section: universe/libs
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Architecture: amd64
Uncompressed Size: 1426 k
Depends: libasound2 (>= 1.0.27), libc6 (>= 2.29), libpulse0 (>= 0.99.1), libwayland-client0 (>= 1.9.91),
         libwayland-cursor0 (>= 1.0.2), libwayland-egl1 (>= 1.15.0), libx11-6 (>= 2:1.2.99.901), libxcursor1 (> 1.1.2),
         libxext6, libxi6 (>= 2:1.2.99.4), libxinerama1, libxkbcommon0 (>= 0.5.0), libxrandr2 (>= 2:1.2.99.3), libxss1,
         libxxf86vm1
Conflicts: libsdl-1.3-0
Replaces: libsdl-1.3-0
Description: Simple DirectMedia Layer
 SDL is a library that allows programs portable low level access to a video framebuffer, audio output, mouse, and
 keyboard.

 This version of SDL is compiled with X11 and Wayland graphics drivers and OSS, ALSA, sndio and PulseAudio sound
 drivers.
Homepage: https://www.libsdl.org/
```

**Comment by kha84** [comment]
Sorry for keeping a dramatic pause guys, got sick in between :)

So here's the dependencies of the SDL lib I have (I wasn't quite sure if **aptitude show** gets this kind of info from the repo or from the local cache, so I checked that also a bit different way). But it's clear it was compiled without X11 support, as nothing indicates the lib got  dependency on any of the X libs:

```
root@orangepi4-lts:~# aptitude show libsdl2-2.0-0
Package: libsdl2-2.0-0            
Version: 2.0.20+5
State: installed
Automatically installed: yes
Multi-Arch: same
Priority: optional
Section: libs
Maintainer: Debian SDL packages maintainers <pkg-sdl-maintainers@lists.alioth.debian.org>
Architecture: arm64
Uncompressed Size: 1,573 k
Depends: libc6 (>= 2.29), libudev1, libdbus-1-3
Conflicts: libsdl-1.3-0
Breaks: libsdl2-2.0-0:armhf (!= 2.0.20+5)
Replaces: libsdl-1.3-0, libsdl2-2.0-0:armhf (< 2.0.20+5)
Description: Simple DirectMedia Layer
 SDL is a library that allows programs portable low level access to a video framebuffer, audio output, mouse, and keyboard. 
 
 This package contains the shared library, compiled with X11 graphics drivers and OSS, ALSA and PulseAudio sound drivers.
Homepage: http://www.libsdl.org/

root@orangepi4-lts:~# apt-rdepends libsdl2-2.0-0
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
libsdl2-2.0-0
  Depends: libc6 (>= 2.29)
  Depends: libdbus-1-3
  Depends: libudev1
libc6
  Depends: libcrypt1
  Depends: libgcc-s1
libcrypt1
  Depends: libc6 (>= 2.25)
libgcc-s1
  Depends: gcc-10-base (= 10.2.1-6)
  Depends: libc6 (>= 2.17)
gcc-10-base
libdbus-1-3
  Depends: libc6 (>= 2.17)
  Depends: libsystemd0
libsystemd0
  Depends: libc6 (>= 2.30)
  Depends: libgcrypt20 (>= 1.8.0)
  Depends: liblz4-1 (>= 0.0~r122)
  Depends: liblzma5 (>= 5.1.1alpha+20120614)
  Depends: libzstd1 (>= 1.4.0)
libgcrypt20
  Depends: libc6 (>= 2.17)
  Depends: libgpg-error0 (>= 1.25)
libgpg-error0
  Depends: libc6 (>= 2.17)
liblz4-1
  Depends: libc6 (>= 2.17)
liblzma5
  Depends: libc6 (>= 2.17)
libzstd1
  Depends: libc6 (>= 2.17)
libudev1
  Depends: libc6 (>= 2.30)

root@orangepi4-lts:~# dpkg -L libsdl2-2.0-0
/.
/usr
/usr/lib
/usr/lib/aarch64-linux-gnu
/usr/lib/aarch64-linux-gnu/libSDL2-2.0.so.0.18.2
/usr/share
/usr/share/doc
/usr/share/doc/libsdl2-2.0-0
/usr/share/doc/libsdl2-2.0-0/BUGS.txt
/usr/share/doc/libsdl2-2.0-0/CREDITS.txt
/usr/share/doc/libsdl2-2.0-0/README-SDL.txt
/usr/share/doc/libsdl2-2.0-0/README.md
/usr/share/doc/libsdl2-2.0-0/changelog.gz
/usr/share/doc/libsdl2-2.0-0/copyright
/usr/lib/aarch64-linux-gnu/libSDL2-2.0.so.0

root@orangepi4-lts:~# ldd /usr/lib/aarch64-linux-gnu/libSDL2-2.0.so.0
	linux-vdso.so.1 (0x0000ffff8a1a1000)
	libm.so.6 => /lib/aarch64-linux-gnu/libm.so.6 (0x0000ffff89f1b000)
	libdl.so.2 => /lib/aarch64-linux-gnu/libdl.so.2 (0x0000ffff89f07000)
	libpthread.so.0 => /lib/aarch64-linux-gnu/libpthread.so.0 (0x0000ffff89ed6000)
	libc.so.6 => /lib/aarch64-linux-gnu/libc.so.6 (0x0000ffff89d60000)
	/lib/ld-linux-aarch64.so.1 (0x0000ffff8a171000)
```

**Comment by nielsmh** [comment]
I think we'll close this since it's a problem with the SDL package from your distribution.

**Comment by glx22** [comment]
Well seems they kept debian description without including what it says it should. (see https://packages.debian.org/bullseye/libsdl2-2.0-0)

**Comment by kha84** [comment]
Yeah, that was very ... disrespectful from their's side :)

I should have paid more attention myself to figure it out, that the package which came with the distro was of **greater** version (2.0.20+5) compared to the one I see in repo (2.0.14+dfsg2-3+deb11u1). This alone should have yelled at me that something is not quite right:

```
root@orangepi4-lts:~# apt list libsdl2-2.0-0 -a
Listing... Done
libsdl2-2.0-0/now 2.0.20+5 arm64 [installed,local]
libsdl2-2.0-0/stable,stable,stable,stable,stable 2.0.14+dfsg2-3+deb11u1 arm64
libsdl2-2.0-0/stable,stable,stable,stable,stable 2.0.14+dfsg2-3+deb11u1 armhf
```


```
root@orangepi4-lts:~# apt policy libsdl2-2.0-0
libsdl2-2.0-0:
  Installed: 2.0.20+5
  Candidate: 2.0.20+5
  Version table:
 *** 2.0.20+5 100
        100 /var/lib/dpkg/status
     2.0.14+dfsg2-3+deb11u1 500
        500 http://deb.debian.org/debian bullseye/main arm64 Packages
        500 http://ftp.ru.debian.org/debian bullseye/main arm64 Packages
        500 https://mirror.docker.ru/debian bullseye/main arm64 Packages
        500 http://ftp.by.debian.org/debian bullseye/main arm64 Packages
        500 http://mirror.cloudhosting.lv/debian bullseye/main arm64 Packages
```

What eventually helped me to resolve this my SDL issue is that I created a backup copy of that lib (just in case if I have to revert back to it) and installed the version from repo on a top of it:
```
# cp /lib/aarch64-linux-gnu/libSDL2-2.0.so.0.18.2 /lib/aarch64-linux-gnu/libSDL2-2.0.so.0.18.2.bak

# apt-get download libsdl2-2.0-0=2.0.14+dfsg2-3+deb11u1
Get:1 http://deb.debian.org/debian bullseye/main arm64 libsdl2-2.0-0 arm64 2.0.14+dfsg2-3+deb11u1 [460 kB]
Fetched 460 kB in 3s (146 kB/s)                              
W: Download is performed unsandboxed as root as file '/root/libsdl2-2.0-0_2.0.14+dfsg2-3+deb11u1_arm64.deb' couldn't be accessed by user '_apt'. - pkgAcquire::Run (13: Permission denied)

# dpkg -i ./libsdl2-2.0-0_2.0.14+dfsg2-3+deb11u1_arm64.deb 
dpkg: warning: downgrading libsdl2-2.0-0:arm64 from 2.0.20+5 to 2.0.14+dfsg2-3+deb11u1
(Reading database ... 214380 files and directories currently installed.)
Preparing to unpack .../libsdl2-2.0-0_2.0.14+dfsg2-3+deb11u1_arm64.deb ...
Unpacking libsdl2-2.0-0:arm64 (2.0.14+dfsg2-3+deb11u1) over (2.0.20+5) ...
Setting up libsdl2-2.0-0:arm64 (2.0.14+dfsg2-3+deb11u1) ...
Processing triggers for libc-bin (2.31-13+deb11u3) ...


# cd /lib/aarch64-linux-gnu/
# unlink libSDL2-2.0.so.0
# ln -s libSDL2-2.0.so.0.14.0  libSDL2-2.0.so.0
```

OpenTTD now works for me, thanks a lot everyone for the help.

Just need to test it now, if that SDL library swapping didn't break anything else in the EmulationStation / RetroArch which were also coming as precompiled with precompiled cores together with this distro.

**Comment by Eilon** [comment]
I had this same error trying to build and run OpenTTD on my MacBook Pro M4. I was able to compile, but then running it produced the same video driver error.

I ran `openttd --help` to see which video drivers were available and it listed these:

```
List of video drivers:
      cocoa-opengl: Cocoa OpenGL Video Driver
         dedicated: Dedicated Video Driver
              null: Null Video Driver
```

Using `./openttd -v dedicated` didn't fail, but it also didn't show anything (just the console/terminal window).

What worked is running `./openttd -v cocoa-opengl`, which worked perfectly.
